components/plugin_marketplace/marketplace_modal.tsx

```diff
@@ -84,81 +90,89 @@ export const InstalledPlugins = ({installedPlugins, changeTab}) => {
     return <MarketplaceList plugins={installedPlugins}/>;
 };
 
-InstalledPlugins.propTypes = {
-    installedPlugins: PropTypes.array.isRequired,
-    changeTab: PropTypes.func,
+export type MarketplaceModalProps = {
+    show: boolean;
+    plugins: MarketplacePlugin[];
+    installedPlugins: MarketplacePlugin[];
+    siteURL: string;
+    pluginStatuses?: Dictionary<PluginStatusRedux>,
+    actions: {
+        closeModal: () => void;
+        fetchPlugins: (localOnly?: boolean) => Promise<{error?: Error}>;
+        filterPlugins(filter: string): Promise<{error?: Error}>;
+    };
+};
+
+type MarketplaceModalState = {
+    tabKey: any;
+    loading: boolean;
+    serverError: Error | undefined;
```

Comment:
You can mark it as `serverError?: Error` instead of `Error | undefined`.

Reply:
Good point. Simplified in 190e535af498cf64500fa74c6cea78fe604c4ad2.